js/packages/openinference-instrumentation-openai/src/instrumentation.ts

```diff
@@ -119,23 +120,27 @@ export class OpenAIInstrumentation extends InstrumentationBase<typeof openai> {
             },
           );
           const wrappedPromise = execPromise.then((result) => {
-            if (result) {
+            if (isChatCompletionResponse(result)) {
               // Record the results
               span.setAttributes({
                 [SemanticConventions.OUTPUT_VALUE]: JSON.stringify(result),
                 [SemanticConventions.OUTPUT_MIME_TYPE]: MimeType.JSON,
                 // Override the model from the value sent by the server
-                [SemanticConventions.LLM_MODEL_NAME]: isChatCompletionResponse(
-                  result,
-                )
-                  ? result.model
-                  : body.model,
+                [SemanticConventions.LLM_MODEL_NAME]: result.model,
                 ...getLLMOutputMessagesAttributes(result),
                 ...getUsageAttributes(result),
               });
+              span.setStatus({ code: SpanStatusCode.OK });
+              span.end();
+            } else {
+              // This is a streaming response
+              // handle the chunks and add them to the span
+              // First split the stream via tee
+              const [leftStream, rightStream] = result.tee();
```

Comment:
Does `tee` create a two-speed scenario? For example:
1. If user cancels the stream, do we go ahead and read the whole thing, incurring additional token costs?
   - What if the stream never ends?
3. If network error happens mid-stream, how does that error bubble up to the user?

Reply:
I'm not fully sure on either - both valid concerns. Right now I'm trying to avoid an extremely convoluted instrumentation. Let me add some tests and get back to you.

Reply:
Sounds good, thanks!